gateway/ln-gateway/src/db.rs

```diff
@@ -508,6 +523,82 @@ async fn migrate_to_v4(mut ctx: MigrationContext<'_>) -> Result<(), anyhow::Erro
     Ok(())
 }
 
+async fn migrate_to_v5(mut ctx: MigrationContext<'_>) -> Result<(), anyhow::Error> {
+    let mut dbtx = ctx.dbtx();
+    migrate_federation_config(&mut dbtx).await
+}
+
+async fn migrate_federation_config(
+    dbtx: &mut DatabaseTransaction<'_>,
+) -> Result<(), anyhow::Error> {
+    async fn migrate_client_entries(
+        dbtx: &mut DatabaseTransaction<'_>,
+        prefix: &[u8],
+    ) -> Result<(), anyhow::Error> {
+        let isolated_entries = dbtx
+            .raw_find_by_prefix(prefix)
+            .await?
+            .collect::<BTreeMap<_, _>>()
+            .await;
+        for (mut key, value) in isolated_entries {
+            dbtx.raw_remove_entry(&key).await?;
+            let mut new_key = vec![DbKeyPrefix::ClientDatabase as u8];
+            new_key.append(&mut key);
+            dbtx.raw_insert_bytes(&new_key, &value).await?;
+        }
+
+        Ok(())
+    }
+
+    // We need to migrate all isolated database entries to be behind the 0x10
+    // prefix. The problem is, if there is a `FederationId` that starts with
+    // 0x04, we cannot read the `FederationId` because the database will be confused
+    // between the isolated DB and the `FederationIdKey` record. To solve this,
+    // we first try and see if there are any entries that begin with
+    // 0x0404. This indicates a joined federation that will have a
+    // problem decoding the ID.
+    let problem_fed_configs = dbtx
+        .raw_find_by_prefix(&[0x04, 0x04])
```

Comment:
Could `0x04` ever be a legitimate `FederationConfig`? I guess it would need to be an invite code with a len 4, which I guess guess is not possible, but just double checking.

Also, in the future this whole migration would not run at all on new databases (which should be empty anyway), so I guess we don't need to worry about it...

Reply:
Yes, exactly

Reply:
I keep thinking ... is this a wrong way to go about it ... ?

Maybe we should all the `0x04` as raw, and use only ones that decode correctly as `FederationConfig` as an id, while ones that don't mark as a prefix to migrate? That would probably kind of read better too and wouldn't be so tricky.

Reply:
I don't know ... Probably current impl works too.

Reply:
As discussed on the call, I re-implemented this using decoding as a filter.